pulsar/producer_impl.go

```diff
@@ -88,12 +88,17 @@ func newProducer(client *client, options *ProducerOptions) (*producer, error) {
 		options.BatchingMaxPublishDelay = defaultBatchingMaxPublishDelay
 	}
 
+	ms, err := client.metrics.GetTopicMetrics(options.Topic)
```

Comment:
This seems like a strange way of testing the topic. I don't think we should rely on calling a metrics function. Why not just call ` internal.ParseTopicName(topic)`. Same for the consumer_impl and tests.

Reply:
The CreateProducer() passes an unchecked topic.And then use GetTopicMetrics() func (Use ParseTopicName func but uncheck err. When use as object transfer internal members ,you will get a runtime panic)

Reply:
I resubmit here

Reply:
I resubmit here

Reply:
If I verify the legitimacy on the outer layer. GetTopicMetrics() passes in *TopicName, is it reasonable? In the same way, the following operations do not need to verify the legitimacy

Reply:
done!